src/lib.rs

```diff
@@ -314,6 +327,20 @@ impl CANSocket {
         Ok(())
     }
 
+    /// Control Flexible Data Rate
+    pub fn set_fd_frames(&self, fd_frames_enable: bool) -> io::Result<()> {
+        let fd_frames_enable = fd_frames_enable as c_int;
+        let opt_ptr = &fd_frames_enable as *const c_int;
+        let rv = unsafe {
+            setsockopt(self.fd, SOL_CAN_RAW, CAN_RAW_FD_FRAMES, opt_ptr as *const c_void, size_of::<c_int>() as u32)
```

Comment:
Will this operation fail when only "normal" CAN is available? Otherwise it might be worthwhile setting this in general on all created CANSockets, might remove some (mildly) sharp corners for API users. Scenario: I open a socket and expect to be able to send FD frames. However, for some reason, ```send_frame()``` keeps failing. 

Since this change already makes the inner type of the CANFrame struct be compatible with CAN-FD (i.e. library users pay the cost for the memory anyway), it might make sense to just set this option in general.

Reply:
It may fail if the controller does not support FD mode, or maybe for other reasons as well.
I don't think setting it automatically is a good idea. Rather let user have control on what's going on and let him set the mode explicitly than let some things happens under the hood.

Your proposal (2.) in your comment https://github.com/socketcan-rs/socketcan-rs/pull/16#issuecomment-1328206813 seems to me like a good approach.